src/OrchardCore/OrchardCore.Abstractions/Shell/ShellSettings.cs

```diff
@@ -29,11 +29,15 @@ public ShellSettings(ShellConfiguration settings, ShellConfiguration configurati
             _configuration = configuration;
         }
 
+        /// <summary>
+        /// This construction should only be used for setup purposes as it will generate the immutable TenantId.
+        /// </summary>
+        /// <param name="settings"></param>
         public ShellSettings(ShellSettings settings)
         {
             _settings = new ShellConfiguration(settings._settings);
             _configuration = new ShellConfiguration(settings.Name, settings._configuration);
-            TenantId = settings.TenantId;
+            TenantId = IdGenerator.GenerateId();
```

Comment:
The passed settings already contains the `TenantId` set as soon as the tenant was created, and that can be preserved.

Reply:
As soon as the tenant was created by the UI, API or WF, also whe the Default tenant is created by code

Reply:
The idea is that I added an Init setter on the TenantId so it now needs to be set from a constructor. And it happens that this constructor is the one used for the setup only. The idea is that we should not be able to change the TenantId. Just use the one from the ShellSettings taken from the json file. Of course, anyone could change that TenantId from that json file afterward but that would be a mistake in that case. What I want is that this TenantId be set on setup and afterward nothing should be able to change it.

Reply:
We may need the TenantId also for an Uninitilized tenant, before the tenant setup is done, e.g. if we add the redis data protection to the Setup Features, or any other feature that would use this TenantId.